extensions/resource-deployment/src/ui/resourceTypeWizard.ts

```diff
@@ -147,10 +147,12 @@ export class ResourceTypeWizard {
 		pages.unshift(new ToolsAndEulaPage(this));
 		this.wizardObject!.pages = pages.map(p => p.pageObject);
 		this.pages = pages;
-		this.pages.forEach((page) => {
+		this.pages.forEach((page, idx) => {
 			page.pageObject.onValidityChanged((isValid: boolean) => {
-				// generateScriptButton is enabled only when the page is valid.
-				this.wizardObject.generateScriptButton.enabled = isValid;
+				if (idx === pages.length - 1) {
```

Comment:
This is assuming that the last page is the only one showing the generate scripts - which may not necessarily be true in the future.

Why isn't each page just setting the button status when it's navigated to based on the current state of its validation?

Reply:
The enabled state of generateScriptButton should just be tied to the validity of the page. The hidden state is used to toggle whether generateScript button is visible or not. Only for NotebookWizards currently, we want the generateScript button to be visible on the last change. In the future, each pageInfo object in the JSON could expose a flag to display generateScriptButton on that page.